Jobs.Fetcher.Twitter/Helpers/DbReader.cs

```diff
@@ -18,49 +18,98 @@ namespace Jobs.Fetcher.Twitter.Helpers {
 
     public static class DbReader {
 
-        public static User GetUserByUsername(string username, DataLakeTwitterDataContext dbContext) {
-            return dbContext.Users.FirstOrDefault(u => u.Username == username);
+        public static User GetUserByUsername(string username, DataLakeTwitterDataContext dbContext, Logger logger) {
+            try {
+                return dbContext.Users.FirstOrDefault(u => u.Username == username);
+            } catch (Exception e) {
+                logger.Error($"Error when looking for User {username} in the database");
+                logger.Verbose($"Error: {e}");
+                return null;
+            }
         }
 
-        public static IEnumerable<LocalAdsAccount> GetAdsAccounts(string username, DataLakeTwitterAdsContext dbContext) {
-            return dbContext.AdsAccounts.Where(a => a.Username == username).ToList();
+        public static IEnumerable<LocalAdsAccount> GetAdsAccounts(string username, DataLakeTwitterAdsContext dbContext, Logger logger) {
+            try {
+                return dbContext.AdsAccounts.Where(a => a.Username == username).ToList();
+            } catch (Exception e) {
+                logger.Error($"Error when looking for AdsAccounts for {username} in the database");
+                logger.Verbose($"Error: {e}");
+                return new List<LocalAdsAccount>();
+            }
         }
 
-        public static IEnumerable<string> GetAdsAccountIds(string username, DataLakeTwitterAdsContext dbContext) {
-            return dbContext.AdsAccounts.Where(a => a.Username == username).Select(a => a.Id).ToList();
+        public static IEnumerable<string> GetAdsAccountIds(string username, DataLakeTwitterAdsContext dbContext, Logger logger) {
+            try {
+                return dbContext.AdsAccounts.Where(a => a.Username == username).Select(a => a.Id).ToList();
+            } catch (Exception e) {
+                logger.Error($"Error when looking for AdsAccountIds for {username} in the database");
+                logger.Verbose($"Error: {e}");
+                return new List<string>();
+            }
         }
 
-        public static IEnumerable<LocalCampaign> GetRunningCampaigns(DataLakeTwitterAdsContext dbContext) {
-            return dbContext.Campaigns.Where(c => c.EndTime != null && c.EndTime > DateTime.UtcNow).ToList();
+        public static IEnumerable<LocalCampaign> GetRunningCampaigns(DataLakeTwitterAdsContext dbContext, Logger logger) {
+            try {
```

Comment:
The considerations I raised for the last functions apply also here and in the next functions